lib/LedgerSMB/Report/Listings/TemplateTrans.pm

```diff
@@ -110,18 +110,14 @@ sub name {
 
 =cut
 
-my %jtype = (
-    1 => 'gl',
-    2 => 'ar',
-    3 => 'ap',
-    );
+my @jtype = ( undef, 'gl', 'ar', 'ap' );    # XXX magic number map.  From where is the magic? --rir
```

Comment:
I agree.  `XXX` is a marker for severely damaged code. 

From whence is this magic?     Or if it is ad hoc,  what shall we call it?